app/assets/javascripts/discourse/components/json-file-uploader.js.es6

```diff
@@ -0,0 +1,105 @@
+
+export default Em.Component.extend({
+  fileInput: null,
+  loading: false,
+  expectedRootObjectName: null,
+  hover: 0,
+
+  classNames: ['json-uploader'],
+
+  _initialize: function() {
+    const $this = this.$();
+    const self = this;
+
+    const $fileInput = $this.find('#js-file-input');
+    this.set('fileInput', $fileInput[0]);
+
+    $fileInput.on('change', function() {
+      self.fileSelected(this.files);
+    });
+
+    const $dragContainer = $this.find('.jsfu-shade-container');
```

Comment:
I tried using the system and found that drag&drop was needed for a nice flow, at least in Chrome.

 1. go to meta, click download button on theme, it goes to bottom of window
 2. go to your site, click import
 3. drag file from the bottom up into the import window